packages/terra-abstract-modal/src/AbstractModal.jsx

```diff
@@ -86,39 +81,75 @@ const defaultProps = {
 class AbstractModal extends React.Component {
   constructor() {
     super();
+    this.state = {
+      modalTrigger: undefined,
+    };
     this.handleKeydown = this.handleKeydown.bind(this);
+    this.showModalDomUpdates = this.showModalDomUpdates.bind(this);
+    this.hideModalDomUpdates = this.hideModalDomUpdates.bind(this);
+    this.modalElement = React.createRef();
   }
 
   componentDidMount() {
     document.addEventListener('keydown', this.handleKeydown);
+
+    if (this.props.isOpen) {
+      this.showModalDomUpdates();
+    }
   }
 
   componentDidUpdate(prevProps) {
-    // When the Modal is no longer in focus, it should no longer listen to the keydown event to handle the Escape key.
-    // Otherwise, the Modal would also get closed when the intention for pressing the Escape key is to close a popup inside the modal.
-    if (!this.props.isFocused && prevProps.isFocused) {
-      document.removeEventListener('keydown', this.handleKeydown);
-    } else if (this.props.isFocused && !prevProps.isFocused) {
-      document.addEventListener('keydown', this.handleKeydown);
+    if (this.props.isOpen) {
+      if (!prevProps.isOpen) {
+        this.showModalDomUpdates();
+      }
+    } else if (prevProps.isOpen) {
+      this.hideModalDomUpdates();
     }
   }
 
   componentWillUnmount() {
     document.removeEventListener('keydown', this.handleKeydown);
-    this.toggleVisuallyHiddenMainDocument('false');
+    this.hideModalDomUpdates();
   }
 
-  handleKeydown(e) {
-    if (e.keyCode === KeyCode.KEY_ESCAPE && this.props.isOpen && this.props.closeOnEsc && this.props.isFocused) {
-      this.props.onRequestClose();
+  showModalDomUpdates() {
+    const mainDocumentElement = document.querySelector(this.props.rootSelector);
+    // Store element that was last focused prior to modal opening
+    this.setState({ modalTrigger: document.activeElement });
+
+    if (mainDocumentElement) {
+      mainDocumentElement.setAttribute('aria-hidden', 'true');
+      mainDocumentElement.setAttribute('inert', '');
+      // Shift focus to modal when opened
+      this.modalElement.current.focus();
     }
   }
 
-  toggleVisuallyHiddenMainDocument(hiddenValue) {
+  hideModalDomUpdates() {
     const mainDocumentElement = document.querySelector(this.props.rootSelector);
 
     if (mainDocumentElement) {
-      mainDocumentElement.setAttribute('aria-hidden', hiddenValue);
+      mainDocumentElement.setAttribute('aria-hidden', 'false');
+      mainDocumentElement.removeAttribute('inert');
+    }
+
+    setTimeout(() => {
```

Comment:
Does this timeout seem to be sufficient for slower browsers like IE10? Would it be sufficient to use an animation frame or a timeout length of 0 to force this to be asynchronous?